src/vanilla/utils/atomWithReset.ts

```diff
@@ -2,7 +2,7 @@ import { atom } from '../../vanilla.ts'
 import type { WritableAtom } from '../../vanilla.ts'
 import { RESET } from './constants.ts'
 
-type SetStateActionWithReset<Value> =
+export type SetStateActionWithReset<Value> =
```

Comment:
It's our coding style not to export internal types. It's a bit tedious, but please define the same type in atomWithDefault.ts.

Reply:
In the past, some people depended on internal types, and when we changed the name of the internal type, they got confused. (It felt like a breaking change.) So, it's for us to freely change the internal types. And, this is due to our build process, which doesn't bundle type files unlike js files.